geode-core/src/main/java/org/apache/geode/internal/size/SizeableObjectSizer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.size;
+
+/**
+ * A sizer that allows for efficient sizing of objects that implement the {@link Sizeable}
+ * interface, and delegates to {@link ReflectionSingleObjectSizer} otherwise.
+ */
+public class SizeableObjectSizer implements SingleObjectSizer {
+
+  private final SingleObjectSizer sizer = new ReflectionSingleObjectSizer();
+
+  @Override
+  public long sizeof(Object object) {
```

Comment:
The `SingleObjectSizer` interface that `SizeableObjectSizer` implements returns a long from its `sizeof()` method, so returning an int here would mean either breaking the inheritance relationship between the two or changing `SingleObjectSizer` to return int instead of long, which will also affect `CachingSingleObjectSizer`, `InstrumentationSingleObjectSizer` and `ReflectionSingleObjectSizer`, with further knock-on effects that are outside the scope of this PR. As a compromise, would it be acceptable to leave the return type as long, but call `Coder.narrowLongToInt()` on the returned value rather than casting it? This will prevent integer overflow in the size, but avoid the need to do refactoring outside the redis module.

Reply:
Couldn't you just change the new SizeableObjectSizer to not implement the interface? It could be in the redis module instead of core. It is currently only used by redis. I think the classes that use it don't need it to be a SingleObjectSizer. I like the idea of it using Coder.narrowLongToInt but would prefer it to be done in one place instead of all the callers.

Reply:
Yeah, that would work. I'll move the Sizer and not have it implement the interface.